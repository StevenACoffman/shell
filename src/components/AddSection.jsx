import React, { Component, PropTypes } from "react";
import { connect } from "react-redux";
import { addSection } from "./../actions";

class AddSection extends Component {
    render() {
        const { dispatch } = this.props;
        return (
            <button
                id="add-section"
                onClick={event => {
                    dispatch(addSection());
                }}
                >
                + Add Section
            </button>
        );
    }
}

AddSection.propTypes = {
    dispatch: PropTypes.func.isRequired
};

AddSection = connect()(AddSection);

export default AddSection;
