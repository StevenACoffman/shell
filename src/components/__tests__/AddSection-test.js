import React from "react";
import AddSection from "../AddSection.jsx";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import {initializeState} from "../../containers/__tests__/outline-test.js";

const mockStore = configureStore([]);
const store = mockStore(initializeState());

it("renders correctly", () => {
    const tree = renderer.create(
        <Provider store={store}>
            <AddSection/>
        </Provider>
  ).toJSON();
    expect(tree).toMatchSnapshot();
});
