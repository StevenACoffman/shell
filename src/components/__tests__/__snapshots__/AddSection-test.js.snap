exports[`test renders correctly 1`] = `
<div>
  <div
    className="row">
    <div>
      <span>
        <label
          htmlFor="citation-format-picker">
          Citation Format:
        </label>
        <select
          id="citation-format-picker"
          onChange={[Function]}
          value="">
          <option
            value="mla">
            MLA
          </option>
          <option
            value="apa">
            APA
          </option>
          <option
            value="chicago">
            CHICAGO
          </option>
        </select>
      </span>
      <ul
        className="list-jstor hide" />
    </div>
  </div>
  <div
    className="row">
    <div
      className="small-12 columns">
      <div
        id="thesis-component">
        <label
          htmlFor="thesis-body">
          Thesis Statement
        </label>
        <textarea
          id="thesis-body"
          onChange={[Function]}
          placeholder="Placeholder..."
          rows="5"
          type="text"
          value="" />
      </div>
      <div>
        <div
          className="row">
          <div
            className="small-9 columns">
            <div>
              <label
                htmlFor="name_0">
                Section Name
              </label>
              <input
                id="name_0"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
            <div>
              <label
                htmlFor="notes_0">
                Section Notes
              </label>
              <textarea
                id="notes_0"
                name="notes"
                onChange={[Function]}
                rows="5"
                value="" />
            </div>
            <div>
              <div>
                <span
                  className="subheader">
                  Citations
                </span>
                <ul
                  className="list-jstor"
                  id="citations_0" />
              </div>
            </div>
            <div>
              <button
                className="btn-link"
                data-reveal-id="sectionModal_0"
                id="add_citations_section_0"
                onClick={[Function]}>
                + Add Citation From List
              </button>
              <noscript />
            </div>
          </div>
          <div
            className="small-3 columns">
            <div
              className="mtxl action-buttons">
              <div>
                <button
                  className="button button-jstor secondary disabled"
                  disabled={true}
                  id="move_section_up_button_0"
                  onClick={[Function]}
                  type="button">
                  Move Section Up
                </button>
              </div>
              <div>
                <button
                  className="button button-jstor secondary disabled"
                  disabled={true}
                  id="move_section_down_button_0"
                  onClick={[Function]}
                  type="button">
                  Move Section Down
                </button>
              </div>
              <button
                className="button button-jstor"
                id="delete_section_0"
                onClick={[Function]}>
                Delete Section
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        className="button button-jstor"
        id="add-section"
        onClick={[Function]}
        type="button">
        + Add Section
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly 2`] = `
<button
  className="button button-jstor"
  id="add-section"
  onClick={[Function]}
  type="button">
  + Add Section
</button>
`;
